Core/Src/util/initializers.c

```diff
@@ -121,3 +122,11 @@ void deInit_all_peripherals() {
     HAL_TIM_Base_DeInit(&htim1);
     HAL_TIM_Base_DeInit(&htim2);
 }
+
+void init_all_ECU(ADC_HandleTypeDef* hadc) {
```

Comment:
n gostei do nome, vc tentou deixar igual ao do deInit, mas la ele realmente dá deinit em todos os periféricos. o init de todos periféricos o cube que faz

```suggestion
void init_ECU() {
```
chame os handle do adc e tim por extern